src/main/scala/wdl4s/cwl/CommandLineTool.scala

```diff
@@ -0,0 +1,97 @@
+package wdl4s.cwl
+
+import eu.timepit.refined._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.string._
+import shapeless.{:+:, CNil}
+import wdl4s.cwl.CwlVersion._
+import wdl4s.cwl.CwlType._
+
+case class CommandLineTool(
```

Comment:
Yeah :) In general though I was just suggesting adding some comments on how this whole approach